azure-devops/azext_devops/dev/team/service_endpoint.py

```diff
@@ -172,3 +199,27 @@ def create_service_endpoint(service_endpoint_configuration,
     import json
     service_endpoint_to_create = json.loads(in_file_content)
     return client.create_service_endpoint(service_endpoint_to_create, project)
+
+
+def update_service_endpoint(id, enable_for_all=None, organization=None,  # pylint: disable=redefined-builtin
+                            project=None, detect=None):
+    """Update a service endpoint
+    :param id: ID of the service endpoint.
+    :type id: str
+    """
+    if enable_for_all is None:
+        raise CLIError('Atleast one property to be updated must be specified.')
+    organization, project = resolve_instance_and_project(detect=detect,
+                                                         organization=organization,
+                                                         project=project)
+    client = get_service_endpoint_client(organization)
+    se = client.get_service_endpoint_details(project, id)
+
+    # set authorization if get service endpoint succeeded
+    from azext_devops.dev.pipelines.pipeline_utils import set_authorize_resource, get_authorize_resource
+    set_authorize_resource(
+        authorized=enable_for_all, res_id=se.id, name=se.name, res_type='endpoint',
+        organization=organization, project=project)
+
+    authorized = get_authorize_resource(res_id=se.id, res_type='endpoint', organization=organization, project=project)
+    return ServiceEndpointAuthorized(service_endpoint_parameters=se, authorized=authorized)
```

Comment:
Why are we doing this?
Can't we just return response from authorize_project_resources ?
Since, this is a separate command for enable-all

Also, if its really necessary can we move ServiceEndpointAuthorized to some helper class?

Reply:
Why a helper class? this is not used anywhere else except with service endpoint commands.

The command is service-endpoint update so it should return details of service endpoint also.. again this is to allow adding more params in future to this command to update actual service endpoint metadata.

Reply:
As discussed offline.. The only concern remains on maintenance overhead of JSON class being referenced here which should be caught in linting check if it changes during client upgrade. 

Let me know if there are any more concerns.